content/en/docs/instrumentation/ruby/sampling.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Sampling
+weight: 6
+---
+
+[Sampling](/docs/concepts/sampling/) is a process that restricts the amount of
+traces that are generated by a system. The Ruby SDK offers several
+[head samplers](/docs/concepts/sampling#head-sampling).
+
+## Default behavior
+
+By default, all spans are sampled, and thus, 100% of traces are sampled. If you
+do not need to manage data volume, don't bother setting a sampler.
+
+Specifically, the default sampler is a composite of [ParentBased][] and
+[ALWAYS_ON][] that ensures the root span in a trace is always sampled, and that
+all child spans respect use their parent's sampling flag to make a sampling
+decision. This guarantees that all spans in a trace are sampled by default.
+
+[ParentBased]:
+  https://www.rubydoc.info/gems/opentelemetry-sdk/OpenTelemetry/SDK/Trace/Samplers/ParentBased
+[ALWAYS_ON]:
+  https://www.rubydoc.info/gems/opentelemetry-sdk/OpenTelemetry/SDK/Trace/Samplers
+
+## TraceIDRatioBased Sampler
+
+The most common head sampler to use is the [TraceIdRatioBased][] sampler. It
+deterministically samples a percentage of traces that you pass in as a
+parameter.
+
+[TraceIdRatioBased]:
+  https://www.rubydoc.info/gems/opentelemetry-sdk/OpenTelemetry/SDK/Trace/Samplers/TraceIdRatioBased
+
+### Environment Variables
+
+You can configure a `TraceIdRatioBased` sampler with environment variables:
+
+```shell
+export OTEL_TRACES_SAMPLER="traceidratio"
+export OTEL_TRACES_SAMPLER_ARG="0.1"
+```
+
+This tells the SDK to sample spans such that only 10% of traces get exported.
+
+### Configuration in Code
+
+You can also configure a `TraceIdRatioBased` sampler in code. Here's an example:
+
+```ruby
+require 'opentelemetry/sdk'
+require 'opentelemetry/exporter/otlp'
+require 'opentelemetry/instrumentation/all'
+
+# sample 10% of traces
+sampling_percentage = 0.1
+sampler = Samplers.trace_id_ratio_based(sampling_percentage)
+
+tracer_provider = OpenTelemetry.tracer_provider(sampler: sampler)
```

Comment:
This example is incorrect. You have to create a new object instance to set on the configurator block below. 

I do not recommend documenting the programmatic configuration for a use case where the users will be setting up default samplers. 

The author would have to initialize the tracer provider themselves and I do not recommend we document it unless folks want to write their own custom samplers: 

https://github.com/open-telemetry/opentelemetry-ruby/blob/main/sdk/lib/opentelemetry/sdk/trace/tracer_provider.rb#L30

Users can do something like this in code to achieve the same results for default samplers

```ruby
ENV['OTEL_TRACES_SAMPLER'] ||= "traceidratio"
ENV['OTEL_TRACES_SAMPLER_ARG'] ||= "0.1"
```

Reply:
Thanks! Yeah, I was worried about this since the Ruby SDK does a great job of letting you easily set things up programmatically, but this one thing all of a sudden sets you down a hard mode path that doesn't feel right. Maybe at some point we can have docs for custom samplers and other stuff, but for now I'm more than happy to nix this whole section.

Reply:
🙇🏼 thank you 🙇🏼